includes/entities/settings-field-arguments.php

```diff
@@ -430,4 +424,93 @@ public function getDefault()
     {
         return $this->default;
     }
+
+    /**
+     * @param mixed $name
+     */
+    public function setName($name): void
+    {
+        $this->name = $name;
+    }
+
+    /**
+     * @param $value
+     */
+    public function setValue($value): void
+    {
+        $this->value = $value;
+    }
+
+    /**
+     * @param string|array $condition
+     *
+     * @return array
+     */
+    private function createCondition($condition): array
+    {
+        if (is_array($condition)) {
+            $parentName  = $this->prefix . $condition['parent_name'];
+            $parentValue = $condition['parent_value'] ?? WCMP_Settings_Data::ENABLED;
+
+            $condition['parents'] = [
+                $parentName => $parentValue,
+            ];
+
+            unset($condition['parent_name']);
+            unset($condition['parent_value']);
+
+            $newCondition = array_replace_recursive(
+                self::CONDITION_DEFAULTS,
+                $condition
+            );
+        } else {
+            $parentName   = $this->prefix . $condition;
+            $newCondition = array_merge(
+                self::CONDITION_DEFAULTS,
+                [
+                    "parents" => [
+                        $parentName => WCMP_Settings_Data::ENABLED,
+                    ],
+                ]
+            );
+        }
+
+        return $newCondition;
+    }
+
+    /**
+     * @param array $conditionData
+     *
+     * @return array
+     */
+    private function mergeConditions(array $conditionData): array
+    {
+        $mergedConditions = [];
+
+        foreach ($conditionData as $condition) {
+            $foundMatch = false;
+
+            foreach ($mergedConditions as &$mergedCondition) {
+                $typeMatches     = $mergedCondition['type'] === $condition['type'];
+                $setValueMatches = $mergedCondition['set_value'] === $condition['set_value'];
+
+                if ($typeMatches && $setValueMatches) {
+                    $mergedCondition['parents'] = array_merge(
+                        $mergedCondition['parents'],
+                        $condition['parents']
+                    );
+
+                    $foundMatch = true;
```

Comment:
Ipv $foundMatch kan je hier toch beter $mergedConditions vullen (met `$mergedConditions[] = $condition;`)?
Dan kan `foreach ($mergedConditions as &$mergedCondition) {` veranderen in `foreach ($mergedConditions as $mergedCondition) {`. Z'n reference wil je eigenlijk altijd zoveel mogelijk vermijden (traag en onduidelijk).